android/store/src/main/java/com/coinbase/wallet/store/Store.kt

```diff
@@ -70,13 +70,19 @@ class Store(context: Context) : StoreInterface {
         deleteAllEntries(kinds = StoreKind.values())
     }
 
-    override fun removeAll(kinds: Array<StoreKind>) = accessLock.write {
-        if (isDestroyed) return
+    @Suppress("UNCHECKED_CAST")
+    override fun removeAll(kinds: Array<StoreKind>) {
+        accessLock.write {
+            if (isDestroyed) return
 
-        deleteAllEntries(kinds = StoreKind.values())
+            deleteAllEntries(kinds)
+        }
 
         changeObservers.values.forEach {
             val observer = it as? BehaviorSubject<Optional<Any>>
+
+            if (isDestroyed) return
```

Comment:
Should this check be inside a write lock?

Reply:
Ooopps. It should be removed from here. Only needed on line 75. fixed